### PR TITLE
SAR-4540 Added service for calculating whether accounting period is v…

### PR DESCRIPTION
**Pull request by MarjanMubarok**

### CODE REVIEW

File: app/core/services/AccountingPeriodService.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package core.services
+
+import java.time.LocalDate
+
+import incometax.util.{AccountingPeriodUtil, CurrentDateProvider}
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class AccountingPeriodService @Inject()(currentDateProvider: CurrentDateProvider) {
+  def accountingPeriodElligibilty(starDate: LocalDate, endDate: LocalDate): Boolean = {
```
Comment (r-melvin) [inline app/core/services/AccountingPeriodService.scala]: could you name this something like isEligible or isEligiblePeriod

File: app/core/services/AccountingPeriodService.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    isWholeYear(starDate, endDate) && isValidTaxYear(starDate, endDate)
+  }
+
+  def isWholeYear(startDate: LocalDate, endDate: LocalDate): Boolean = {
```
Comment (r-melvin) [inline app/core/services/AccountingPeriodService.scala]: Do these methods need to be exposed or can they be private?

File: app/core/services/AccountingPeriodService.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    startDate.minusDays(1).plusYears(1) == endDate
+  }
+
+  def isValidTaxYear(startDate: LocalDate, endDate: LocalDate): Boolean = {
+    val currentTaxYearEndYear = AccountingPeriodUtil.getTaxEndYear(currentDateProvider.getCurrentDate())
```
Comment (r-melvin) [inline app/core/services/AccountingPeriodService.scala]: val endOfCurrentTaxYear?

File: app/core/services/AccountingPeriodService.scala (hunk 2 above)
Comment (MarjanMubarok) [reply app/core/services/AccountingPeriodService.scala]: they are no longer separate and have been moved into the method.

**Pull request merged**